src/components/Input.react.js

```diff
@@ -24,294 +24,297 @@ function formatValue(val, type){
  */
- export default class Input extends Component {
-	constructor(props) {
-		super(props);
-		this.state = {
-			value: formatValue(props.value, props.type)
-		};
-	}
-
-  componentWillReceiveProps(nextProps) {
-		this.setState({value: formatValue(nextProps.value, nextProps.type)});
-	}
-
-	render() {
-		const {
-      fireEvent,
-      setProps,
-      type,
-      updatemode
-    } = this.props;
-
-		const {value} = this.state;
-		return (
-      <input
-        onBlur={
-          () => {
-            const newValue = formatValue(this.state.value, type)
-            if (updatemode === 'blur') {
-              this.setState({value: newValue});
-              if (setProps) {
-                  setProps({value: newValue});
-              }
-              if (fireEvent) {fireEvent({event: 'blur'});}
-            }
-          }
-        }
-        onChange={
-          (e) => {
-            const newValue = formatValue(e.target.value, type)
-            this.setState({value: newValue});
-    				if (updatemode === 'keypress') {
-              if (setProps) {
-                  setProps({value: newValue});
-              }
-    				}
-    				if (fireEvent) {fireEvent({event: 'change'});}
-  				}
-        }
-        onKeyPress={
-          (e) => {
-            if (type === 'number'){
-              const curVal = this.state.value;
-              const keyCode = e.keyCode || e.which;
-              const keyValue = String.fromCharCode(keyCode)
-              const unicodePlus = 43;
-              const unicodeNine = 57;
-            if ((keyCode <= unicodePlus || keyCode > unicodeNine) || (keyValue === '/'))
-              {e.preventDefault();}
-            if (( keyValue === '-') && (curVal.length > 0))
-              {e.preventDefault();}
-            if ((keyValue === '.' || keyValue === ',') && (/[.]|,/.test(curVal.value)))
-              {e.preventDefault();}
-            }
-          }
-        }
-        onClick = {
-          (e) => {
-            const newValue = formatValue(e.target.value, type)
-            this.setState({value: newValue});
-            if (setProps) {
-              setProps({value: newValue});
-            }
-          }
-        }
-        value={value}
-        {...omit(['fireEvent', 'setProps', 'value', 'updatemode'], this.props)}
-        />
-    );
-	}
+export default class Input extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            value: formatValue(props.value, props.type),
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setState({value: formatValue(nextProps.value, nextProps.type)});
+    }
+
+    render() {
+        const {fireEvent, setProps, type, updatemode} = this.props;
+
+        const {value} = this.state;
+        return (
+            <input
+                onBlur={() => {
+                    const newValue = formatValue(this.state.value, type);
+                    if (updatemode === 'blur') {
```

Comment:
Here, if `updatemode` is an array (i.e. `['blur', 'enter']`) this will not trigger. The best way to do this would probably be to first see if `updatemode` is an array (you could use Ramda's `is` function for that, something like `R.is(Array, updatemode)`) and then check if that array contains `blur` (Ramda has a [contains](https://ramdajs.com/docs/#contains) function that can do that easily)

Reply:
Or maybe even `if (updatemode === 'blur' || R.contains('blur', updatemode)) {}` will work!

Reply:
actually I implemented with `updatemode` being `blur` (update when pressing tab or clicking away) or `keypress` (update at everykeystrokes), I wasn't sure the if we really need to have `['blur', 'enter']` but I kept it in the `Input.Props`